src/masternodes/mn_rpc.cpp

```diff
@@ -3081,39 +3081,288 @@ UniValue isappliedcustomtx(const JSONRPCRequest& request) {
+            CScript ownerScript = DecodeScript(addresses[i]["address"].get_str());
+            CBalances accountBalances;
+            pcustomcsview->ForEachBalance([&](CScript const & owner, CTokenAmount const & balance) {
+                if (owner != ownerScript) return false;
+                accountBalances.Add(balance);
+            }, BalanceKey{ownerScript, 0});
+            walletAccounts.emplace(ownerScript, accountBalances);
+        }
+    }
+
+    return walletAccounts;
+}
+
+typedef enum {
+    // selecting accounts without sorting
+    SelectionForward,
+    // selecting accounts by ascending of sum token amounts
+    // it means that we select first accounts with min sum of
+    // neccessary token amounts
+    SelectionCrumbs,
+    // selecting accounts by descending of sum token amounts
+    // it means that we select first accounts with max sum of
+    // neccessary token amounts
+    SelectionPie,
+} AccountSelectionMode;
+
+static AccountSelectionMode ParseAccountSelectionParam(const std::string selectionParam) {
+    if (selectionParam == "forward") {
+        return SelectionForward;
+    }
+    else if (selectionParam == "crumbs") {
+        return SelectionCrumbs;
+    }
+    else if (selectionParam == "pie") {
+        return SelectionPie;
+    }
+    else {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalide accounts selection mode.");
+    }
+}
+
+std::map<CScript, CBalances> SelectAccountsByTargetBalances(const std::map<CScript, CBalances>& accounts, const CBalances& targetBalances, AccountSelectionMode selectionMode) {
+    std::map<CScript, CBalances> selectedAccountsBalances;
+    std::vector<std::pair<CScript, CBalances>> foundAccountsBalances;
+    CBalances residualBalances(targetBalances);
+
+    // iterate at all accounts to finding all accounts with neccessaru token balances
+    for (auto accIt = accounts.begin(); accIt != accounts.end(); accIt++) {
+        // selectedBalances accumulates overlap between account balances and residual balances
+        CBalances selectedBalances;
+        // iterate at residual balances to find neccessary tokens in account
+        for (auto balIt = residualBalances.balances.begin(); balIt != residualBalances.balances.end(); balIt++) {
+            // find neccessary token amount from current account
+            auto foundTokenAmount = accIt->second.balances.find(balIt->first);
+            // account balance has neccessary token
+            if (foundTokenAmount != accIt->second.balances.end()) {
+                // add token amount to selected balances from current account
+                selectedBalances.Add(CTokenAmount{foundTokenAmount->first, foundTokenAmount->second});
+            }
+        }
+        if (!selectedBalances.balances.empty()) {
+            // added account and selected balances from account to selected accounts
+            foundAccountsBalances.emplace_back(accIt->first, selectedBalances);
+        }
+    }
```

Comment:
Use iterators when you want to delete or you need no simple for, otherwise use range based for, it's more clear.